### PR TITLE
Node api

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

Implement Node API : Time, GetConfig, Help and NewNode..
Majority of the tests in `perunnode` package are integration tests.
See test section for command to run the tests.

##### Category

Implementation Task

##### Relevant issue

Relates to #83
##### Steps to run the tests

1. Start the ganache-cli node:
```
ganache-cli -b 1 --account="0x1fedd636dbc7e8d41a0622a2040b86fea8842cef9d4aa4c582aad00465b7acff,100000000000000000000" --account="0xb0309c60b4622d3071fad3e16c2ce4d0b1e7758316c187754f4dd0cfb44ceb33,100000000000000000000"
```
2. Run the tests with integration flags from the project root folder.
```
go test -tags=integration -count=1 -cover ./...
```
3.  

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

File: cmd/perunnode/node.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	return []string{"payment"}
+}
+
+// OpenSession opens and session on this node using the given config and returns the
+// sessionID that can be used to access this session in subsequent calls.
+func (n *node) OpenSession(configFile string) (string, error) {
+	return "", nil
+}
+
+// GetSession is an special call that should be used internally to get the session instance.
```
Comment (anagha-ks) [inline cmd/perunnode/node.go]:
```suggestion
// GetSession is a special call that should be used internally to get the session instance.
```

File: cmd/perunnode/node.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	n.Logger.Debug("Received request: node.GetConfig")
+	return n.cfg
+}
+
+// Help returns the set of APIs offered by the node.
+func (n *node) Help() []string {
+	return []string{"payment"}
+}
+
+// OpenSession opens and session on this node using the given config and returns the
```
Comment (anagha-ks) [inline cmd/perunnode/node.go]:
```suggestion
// OpenSession opens and session on this node uses the given config and returns the
```

File: cmd/perunnode/node.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+// sessionID that can be used to access this session in subsequent calls.
+func (n *node) OpenSession(configFile string) (string, error) {
+	return "", nil
+}
+
+// GetSession is an special call that should be used internally to get the session instance.
+// This should not exposed to the user.
```
Comment (anagha-ks) [inline cmd/perunnode/node.go]:
```suggestion
// This should not be exposed to the user.
```

File: errors.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package perun
+
+import (
+	"errors"
+)
+
+// APIError represents the errors that will communicated via the user API.
```
Comment (anagha-ks) [inline errors.go]:
```suggestion
// APIError represents the errors that will be communicated via the user API.
```

File: errors.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		return nil
+	}
+	var apiErr APIError
+	if !errors.As(err, &apiErr) {
+		return ErrInternalServer
+	}
+	return apiErr
+}
+
+// Setinal Error values that are relevant for the end user of the node.
```
Comment (anagha-ks) [inline errors.go]:
```suggestion
// Sentinal Error values that are relevant for the end user of the node.
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

**Comment by manoranjith** [comment]
Updated suggestions.

**Review by anagha-ks — APPROVED** [review]

**Pull request merged**